### DISCUSSION TITLE
Cloning a device with the same installed modules

### DISCUSSION THREAD

**Discussion by lucatortiglione** [netbox-community/netbox#12957] [Ideas]
Hi,
if I clone a device already configured with modules i need,
i get the new one without modules.

What about to implement it ?

**Comment by ryanmerolle** [comment]
It is not a feature currently.  Feel free to propose some new clone function where modules are clone as well.

**Comment by martinum4** [comment]
I don't have the correct words for it, but some kind of "collection" for devicetypes that nests them would be nice.
Example-Problem: You order a lot of two different server configurations (but same base model) from one vendor, one contains better PSUs and accelerator cards (which you want to track as inventory item since they cost you quite a lot).
You then would create two "sets", one containing the basic server and normal PSUs and some kind of order-identifier and the other contains the base server, the better PSUs and your accelerator cards.
Rest of the process when importing works just as normal, but instead of selecting the base-server devicetype and populating them afterwars you get to select the new collection-sets instead.
